modules/navier_stokes/test/tests/finite_volume/cns/stagnation_inlet/supersonic_nozzle_hllc.i

```diff
@@ -0,0 +1,315 @@
+  [momentum_y_stagnation_inflow]
+    type = CNSFVHLLCMomentumStagnationInletBC
+    variable = rho_v
+    momentum_component = y
+    fp = fp
+    stagnation_pressure = ${stagnation_pressure}
+    stagnation_temperature = ${stagnation_temperature}
+    boundary = left
+  [../]
+
+  [fluid_energy_stagnation_inflow]
+    type = CNSFVHLLCFluidEnergyStagnationInletBC
+    variable = rho_E
+    fp = fp
+    stagnation_pressure = ${stagnation_pressure}
+    stagnation_temperature = ${stagnation_temperature}
+    boundary = left
+  []
+
+  ## outflow implicit conditions
+  [mass_outflow]
+    type = CNSFVHLLCMassImplicitBC
+    variable = rho
+    fp = fp
+    boundary = right
+  []
+
+  [momentum_x_outflow]
+    type = CNSFVHLLCMomentumImplicitBC
+    variable = rho_u
+    momentum_component = x
+    fp = fp
+    boundary = right
+  []
+
+  [momentum_y_outflow]
+    type = CNSFVHLLCMomentumImplicitBC
+    variable = rho_v
+    momentum_component = y
+    fp = fp
+    boundary = right
+  []
+
+  [fluid_energy_outflow]
+    type = CNSFVHLLCFluidEnergyImplicitBC
+    variable = rho_E
+    fp = fp
+    boundary = right
+  []
+
+  # wall conditions
+  [momentum_x_pressure_wall]
+    type = CNSFVMomImplicitPressureBC
+    variable = rho_u
+    momentum_component = x
+    boundary = wall
+  []
+
+  [momentum_y_pressure_wall]
+    type = CNSFVMomImplicitPressureBC
+    variable = rho_v
+    momentum_component = y
+    boundary = wall
+  []
+[]
+
+[AuxVariables]
+  [Ma]
+    family = MONOMIAL
+    order = CONSTANT
+  []
+
+  [Ma_layered]
+    family = MONOMIAL
+    order = CONSTANT
+  []
+[]
+
+[UserObjects]
+  [layered_Ma_UO]
+    type = LayeredAverage
+    variable = Ma
+    num_layers = 100
+    direction = x
+  []
+[]
+
+[AuxKernels]
+  [Ma_aux]
+    type = NSMachAux
+    variable = Ma
+    fluid_properties = fp
+    use_material_properties = true
+  []
+
+  [Ma_layered_aux]
+    type = SpatialUserObjectAux
+    variable = Ma_layered
+    user_object = layered_Ma_UO
+  []
+[]
+
+[Materials]
+  [var_mat]
+    type = ConservedVarValuesMaterial
+    rho = rho
+    rhou = rho_u
+    rhov = rho_v
+    rho_et = rho_E
+  []
+
+  [fluid_props]
+    type = GeneralFluidProps
+    porosity = 1
+    characteristic_length = 1
+  []
+
+  [sound_speed]
+    type = SoundspeedMat
+    fp = fp
+  []
+[]
+
+[Postprocessors]
+  [cfl_dt]
+    type = ADCFLTimeStepSize
+    c_names = 'sound_speed'
+    vel_names = 'speed'
+    CFL = 0.5
+  []
+
+  [outflow_Ma]
+    type = SideAverageValue
+    variable = Ma
+    boundary = right
+  []
+[]
+
+[Preconditioning]
+  [smp]
+    type = SMP
+    full = true
+    petsc_options_iname = '-pc_type'
+    petsc_options_value = 'lu'
+  []
+[]
+
+[Executioner]
+  type = Transient
+  end_time = 0.1
+
+  [TimeIntegrator]
+    type = ExplicitSSPRungeKutta
+    order = 2
+  []
+  l_tol = 1e-8
+
+  [TimeStepper]
+    type = PostprocessorDT
+    postprocessor = cfl_dt
+  []
+[]
+
+[VectorPostprocessors]
+  [Ma_layered]
+    type = LineValueSampler
```

Comment:
ugh a LineValueSampler sampling on edges and nodes and my PR to fix that never went in

Reply:
Is it still languishing in an open state?

Reply:
yup. Got to find what s up with bison on it